pkg/quota/v1/evaluator/core/pods.go

```diff
@@ -149,6 +149,9 @@ func (p *podEvaluator) Handles(a admission.Attributes) bool {
 	if op == admission.Create {
 		return true
 	}
+	if feature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && op == admission.Update {
```

Comment:
What was happening previously for container image updates? Pod spec isn't 100% immutable so I'm surprised to see this here.

Reply:
Image update does not alter NS quota so evaluator didn't need to know about it .. until now.

Reply:
I am only looking at the first commit right now to focus my comments, but I expected changes to `PodUsageFunc` to look at `ResourcesAllocated` and take the max of either `spec.resources` or `status.resourcesAllocated`.  Let's discuss in SIG Node this week to make sure we are on the same page, but quota should be charging based on the max(resources, resourcesAllocated) which will naturally result in a drop in usage charged when kubelet is reclaiming the resource.

Reply:
I'm just not able to recall the context of our discussion and that resulted in this decision. So on second thoughts, I will keep this at the more conservative Max(). We can relax it in the future if desired.

Reply:
Evaluator code updated to use  Max(requests,allocated). please review.

Reply:
@vinaykul unless I am missing something, the linked comment was unrelated to resource quota.

resource quota should prevent gamification.

namespace has quota for 2Gi

user creates a pod A with request[cpu=1, memory=1Gi]
user creates a pod B with request[cpu=1, memory=1Gi]
user updates pod A with request[cpu=1, memory=100Mi]
pod A is never able to release 900Mi
user is able to create a pod C with request[cpu=1, memory=900Mi]

thanks for updating to the more conservative approach to take the max wrt to quota evaluation!